File: 04_implementation_and_conformance/service_and_sdk_candidates/sdk_contracts_v0_1/typescript/examples/golden_path.ts
import type { OFARMClient } from "../src/ofarm-client-shape";

// Candidate example only. Shows safe sequencing, not production code.
export async function goldenPath(client: OFARMClient) {
  const capabilities = await client.capabilities.get();

  const dryRun = await client.commits.dryRun({
    commitClass: "INTERVENTION_INTENT",
    payloadRef: "example:payload:prescription"
  });

  const submitted = await client.commits.submit(
    {
      commitClass: "INTERVENTION_INTENT",
      payloadRef: "example:payload:prescription"
    },
    {
      idempotencyKey: "example-prescription-001",
      humanApprovalRef: "approval:example:001"
    }
  );

  const query = await client.queries.execute({
    querySpecRef: "example:query:field-treatment-history",
    requireEvidence: true
  });

  const preview = await client.assemblies.dryRun({
    outputType: "PassportView",
    basisRefs: ["field:7"],
    requireFreshness: true
  });

  return { capabilities, dryRun, submitted, query, preview };
}

File: 04_implementation_and_conformance/service_and_sdk_candidates/sdk_contracts_v0_1/python/examples/golden_path.py
from ofarm_client_shape import OFARMClient


def golden_path(client: OFARMClient):
    capabilities = client.capabilities_get()

    dry_run = client.commits_dry_run({
        "commitClass": "INTERVENTION_INTENT",
        "payloadRef": "example:payload:prescription"
    })

    submitted = client.commits_submit(
        {
            "commitClass": "INTERVENTION_INTENT",
            "payloadRef": "example:payload:prescription"
        },
        idempotency_key="example-prescription-001",
        human_approval_ref="approval:example:001",
    )

    query = client.queries_execute({
        "querySpecRef": "example:query:field-treatment-history",
        "requireEvidence": True
    })

    preview = client.assemblies_dry_run({
        "outputType": "PassportView",
        "basisRefs": ["field:7"],
        "requireFreshness": True
    })

    return {
        "capabilities": capabilities,
        "dry_run": dry_run,
        "submitted": submitted,
        "query": query,
        "preview": preview,
    }
